bin/auto-update/git.ts

```diff
@@ -0,0 +1,230 @@
+import { Octokit } from '@octokit/rest';
+
+export class Git {
+  readonly #sh: SH;
+  readonly #settings: Settings;
+
+  constructor(sh: SH, settings: Settings) {
+    this.#sh = sh;
+    this.#settings = settings;
+  }
+
+  openPRIfItDoesntExist = async (gapiTypeName: string): Promise<void> => {
+    const {
+      user,
+      auth,
+      thisRepo,
+      dtRepoOwner: owner,
+      dtRepoName: repo,
+    } = this.#settings;
+    const octokit = new Octokit({
+      auth,
+      userAgent: `${user}/${thisRepo}`,
+      timeZone: 'UTC',
+    });
+
+    console.log(`Opening PR for ${gapiTypeName}...`);
+
+    try {
+      await octokit.pulls.create({
+        owner,
+        repo,
+        title: `[${gapiTypeName}] automatic update`,
+        head: `${user}:${gapiTypeName}`,
+        base: 'master',
+        // TODO: check for updates in the template (using retrieveCommunityProfileMetrics from black-panther-preview, maybe) but it doesn't seem to be updated often
+        body: `
+  - [x] Use a meaningful title for the pull request. Include the name of the package modified.
+  - [x] Test the change in your own code. (Compile and run.)
+  - [x] Add or edit tests to reflect the change. (Run with \`npm test\`.)
+  - [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
+  - [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
+  - [x] Run \`npm run lint package-name\` (or \`tsc\` if no \`tslint.json\` is present).
+
+  Select one of these and delete the others:
+
+  If changing an existing definition:
+  - [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/Maxim-Mazurok/google-api-typings-generator#google-api-typings-generator
+  - [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
+  - [x] If you are making substantial changes, consider adding a \`tslint.json\` containing \`{ "extends": "dtslint/dt.json" }\`. If for reason the any rule need to be disabled, disable it for that line using \`// tslint:disable-next-line [ruleName]\` and not for whole package so that the need for disabling can be reviewed.
+
+  ----
+
+  Please, note: this PR was created automatically by [Maxim-Mazurok/google-api-typings-generator](https://github.com/Maxim-Mazurok/google-api-typings-generator).
+  Types are generated automatically from [Google API Discovery Service](https://developers.google.com/discovery).
+  Before submitting this PR, types were linted and tested automatically.
+  Alternatively, you can [use](https://github.com/Maxim-Mazurok/google-api-typings-generator/issues/85#issuecomment-601133279) these typings from our [\`types\` branch](https://github.com/Maxim-Mazurok/google-api-typings-generator/tree/types) which gets updated, linted and tested every hour.
```

Comment:
I think we should be straight forward about this. Because I'll be opening these PRs from my account and there would be no way to tell if it was opened manually, or automatically I think we should add this message to make it clear how this PR was opened so that maintainers won't get confused if script starts to act weird.
So, my main message is that we should follow PR template and rules (as it was opened by human) but also make it clear that it's automated PR.
As soon as typings ownership will be mine - I we will skip manual review process, so maintainers won't even read this.
If you want to add/remove something, or change wording - I'm all ears, but I want to leave this message in some form.

Reply:
Yeah sorry, I was just referring to the one specific line about downloading the typings from our `types` branch. I'm not sure that's relevant for DT reviewers.

Happy to read through all of this again before we merge to ensure it makes sense!

Reply:
Got it :)
Well, I left this line there so that if PR isn't getting merged for too long and somebody who needs to use the most recent typing finds this PR - he will know that there's this option.
After #90 is done I'll change this line.